src/vivarium_public_health/metrics/disability.py

```diff
@@ -112,31 +101,62 @@ def get_disability_weight_pipeline(self, builder: Builder) -> Pipeline:
     # Aggregators #
     ###############
 
-    def disability_weight_aggregator(self, dw: pd.DataFrame) -> float:
-        return (dw * to_years(self.step_size)).sum().squeeze()
+    def disability_weight_aggregator(self, dw: pd.DataFrame) -> Union[float, pd.Series]:
+        aggregated_dw = (dw * to_years(self.step_size)).sum().squeeze()
+        if isinstance(aggregated_dw, pd.Series):
+            aggregated_dw.index.name = "cause_of_disability"
+        return aggregated_dw
 
     ##################
     # Report methods #
     ##################
 
     def write_disability_results(
         self,
-        cause_state: Optional[DiseaseState],
         measure: str,
         results: pd.DataFrame,
     ) -> None:
-        """Combine each observation's results and save to a single file"""
-
-        kwargs = {
-            "entity_type": cause_state.cause_type if cause_state else "cause",
-            "entity": cause_state.model if cause_state else "all_causes",
-            "sub_entity": cause_state.state_id if cause_state else None,
-            "results_dir": self.results_dir,
-            "random_seed": self.random_seed,
-            "input_draw": self.input_draw,
-        }
-        write_dataframe_to_parquet(
+        """Format dataframe and write out. Note that ylds are unique in that we
+        can't stratify by cause of disability (because there can be multiple at
+        once), and so the results here are actually wide by disability weight
+        pipeline name.
+        """
+
+        # Drop the unused 'value' column and rename the pipeline names to causes
+        results.drop(columns=["value"], inplace=True)
+        results.rename(columns={"disability_weight": "all_causes"}, inplace=True)
+        results.rename(
+            columns={col: col.replace(".disability_weight", "") for col in results.columns},
+            inplace=True,
+        )
+
+        # Stack the causes of disability
+        idx_names = list(results.index.names)
+        results = pd.DataFrame(results.stack(), columns=[COLUMNS.VALUE])
+        # Name the new index level
+        idx_names += [COLUMNS.SUB_ENTITY]
+        results.index.set_names(idx_names, inplace=True)
```

Comment:
Same here

Reply:
I don't think you can chain here b/c one is working on the index and the other is working on the actual df

Reply:
Ah yes, I missed that.